diff --git a/app/build.gradle b/app/build.gradle
index f3b866a5..b47e1ffd 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -148,7 +148,7 @@ dependencies {
     implementation 'androidx.vectordrawable:vectordrawable:1.1.0'
     debugImplementation "com.squareup.leakcanary:leakcanary-android:${versions.leak_canary}"
     releaseImplementation "com.squareup.leakcanary:leakcanary-android-no-op:${versions.leak_canary}"
-    implementation 'com.viewpagerindicator:library:2.4.1@aar'
+    implementation 'fr.avianey.com.viewpagerindicator:library:2.4.1.1@aar'
     implementation "com.commit451:PhotoView:${versions.photoview}"
     implementation 'com.google.code.findbugs:jsr305:3.0.2'
     implementation 'de.greenrobot:eventbus:2.4.0'
diff --git a/build.gradle b/build.gradle
index d88b21f0..53e33b27 100644
--- a/build.gradle
+++ b/build.gradle
@@ -15,9 +15,10 @@ buildscript {
 allprojects {
     repositories {
         google()
-        maven { url "http://dl.bintray.com/populov/maven" }
         jcenter()
-        maven {url "http://oss.sonatype.org/content/repositories/snapshots"}
+        maven {
+            url "https://oss.sonatype.org/content/repositories/snapshots/"
+        }
     }
 
 }
