diff --git a/build.gradle b/build.gradle
index 8829c37..5855978 100644
--- a/build.gradle
+++ b/build.gradle
@@ -2,7 +2,6 @@
 
 buildscript {
     repositories {
-        maven { url 'https://maven.aliyun.com/repository/public' }
         google()
         jcenter()
 
@@ -18,7 +17,6 @@ buildscript {
 
 allprojects {
     repositories {
-        maven { url 'https://maven.aliyun.com/repository/public' }
         google()
         jcenter()
 