diff --git a/app/build.gradle b/app/build.gradle
index c5debdc..a2a8512 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -1,10 +1,5 @@
 apply plugin: 'com.android.application'
 
-// Load keystore
-def keystorePropertiesFile = rootProject.file("keystore.properties");
-def keystoreProperties = new Properties()
-keystoreProperties.load(new FileInputStream(keystorePropertiesFile))
-
 
 android {
     signingConfigs {
