diff --git a/.gitignore b/.gitignore
index 01e00f3..61b43b0 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1 +1,5 @@
+/build-android-release
+/build
+
+
 CMakeLists.txt.user
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 991cd11..54ea064 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,11 +42,11 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 find_package(Doxygen)
 find_package(Git REQUIRED)
-find_package(Qt5 5.14 COMPONENTS Concurrent Core Gui LinguistTools Positioning Quick QuickWidgets Sql Svg REQUIRED)
+find_package(Qt5 5.15 COMPONENTS Concurrent Core Gui LinguistTools Positioning Quick QuickWidgets Sql Svg REQUIRED)
 if( ANDROID )
-    find_package(Qt5 5.14 COMPONENTS AndroidExtras REQUIRED)
+    find_package(Qt5 5.15 COMPONENTS AndroidExtras Concurrent Core Gui LinguistTools Positioning Quick QuickWidgets Sql Svg Qml Location QuickControls2 SerialPort Sensors QmlModels QuickShapes RemoteObjects QmlWorkerScript QuickTemplates2 QuickTest Test Multimedia MultimediaQuick REQUIRED)
 else()
-    find_package(Qt5 5.14 COMPONENTS Widgets REQUIRED)
+    find_package(Qt5 5.15 COMPONENTS Widgets REQUIRED)
 endif()
 
 
diff --git a/fastlane/metadata/android/en-US/full_description.txt b/fastlane/metadata/android/en-US/full_description.txt
new file mode 100644
index 0000000..77b701e
--- /dev/null
+++ b/fastlane/metadata/android/en-US/full_description.txt
@@ -0,0 +1,28 @@
+Enroute is a free flight navigation app for VFR pilots. Designed to be simple,
+functional and elegant, it takes the stress out of your next flight. The program
+has been written by flight enthusiasts, as a project of Akaflieg Freiburg, a
+flight club based in Freiburg, Germany.
+Enroute features a moving map, similar in style to the official ICAO maps.
+Your current position and your flight path for the next five minutes are marked,
+and so is your intended flight route. A simple tap on the display gives you all
+the information about airspaces, airfields and navaids – complete with
+frequencies, codes, elevations and runway information.
+Our free aeronautical maps can be downloaded for offline use. In addition to
+airspaces, airfields and navaids, selected maps also show traffic circuits as
+well as flight procedures for control zones. The maps receive near-weekly
+updates and cover large parts of the world.
+Enroute supports basic flight planning. It allows you to quickly and easily
+compute distances, courses and headings, and gives you an estimate for flight
+time and fuel consumption. If the weather turns bad, the app will show you the
+closest airfields for landing, complete with distances, directions, runway
+information and frequencies.
+The original author strongly supports the provision of the app via f-droid.
+However, this is an unofficial version of the app, which has not been tested
+by the original app author. It might differ from the official version, even if
+the version number agrees with that of the official app.
+Disclaimer: This free app is published in the hope that it might be useful as
+an aid to prudent navigation. It comes with no guarantees. It may not work as
+expected. The data does not come from official sources and might be incomplete,
+outdated or otherwise incorrect. This app is no substitute for proper flight
+preparation or good pilotage. Relying on it as a primary means of navigation
+is most likely illegal, most certainly stupid and potentially suicidal.
diff --git a/fastlane/metadata/android/en-US/images/featureGraphic.png b/fastlane/metadata/android/en-US/images/featureGraphic.png
new file mode 120000
index 0000000..30876af
--- /dev/null
+++ b/fastlane/metadata/android/en-US/images/featureGraphic.png
@@ -0,0 +1 @@
+../../../../../propaganda/Google-Play-Store/featureGraphic.png
\ No newline at end of file
diff --git a/fastlane/metadata/android/en-US/images/icon.png b/fastlane/metadata/android/en-US/images/icon.png
new file mode 120000
index 0000000..02cee08
--- /dev/null
+++ b/fastlane/metadata/android/en-US/images/icon.png
@@ -0,0 +1 @@
+../../../../../generatedSources/metadata/de.akaflieg_freiburg.enroute.png
\ No newline at end of file
diff --git a/fastlane/metadata/android/en-US/images/phoneScreenshots/1.png b/fastlane/metadata/android/en-US/images/phoneScreenshots/1.png
new file mode 120000
index 0000000..c324156
--- /dev/null
+++ b/fastlane/metadata/android/en-US/images/phoneScreenshots/1.png
@@ -0,0 +1 @@
+../../../../../../propaganda/Google-Play-Store/phone-00-Flight.png
\ No newline at end of file
diff --git a/fastlane/metadata/android/en-US/images/phoneScreenshots/10.png b/fastlane/metadata/android/en-US/images/phoneScreenshots/10.png
new file mode 120000
index 0000000..f94db88
--- /dev/null
+++ b/fastlane/metadata/android/en-US/images/phoneScreenshots/10.png
@@ -0,0 +1 @@
+../../../../../../propaganda/Google-Play-Store/tab-13-Info.png
\ No newline at end of file
diff --git a/fastlane/metadata/android/en-US/images/phoneScreenshots/11.png b/fastlane/metadata/android/en-US/images/phoneScreenshots/11.png
new file mode 120000
index 0000000..eb86bf3
--- /dev/null
+++ b/fastlane/metadata/android/en-US/images/phoneScreenshots/11.png
@@ -0,0 +1 @@
+../../../../../../propaganda/Google-Play-Store/tab-14-Route.png
\ No newline at end of file
diff --git a/fastlane/metadata/android/en-US/images/phoneScreenshots/12.png b/fastlane/metadata/android/en-US/images/phoneScreenshots/12.png
new file mode 120000
index 0000000..906d69b
--- /dev/null
+++ b/fastlane/metadata/android/en-US/images/phoneScreenshots/12.png
@@ -0,0 +1 @@
+../../../../../../propaganda/Google-Play-Store/tab-15-NearAD.png
\ No newline at end of file
diff --git a/fastlane/metadata/android/en-US/images/phoneScreenshots/2.png b/fastlane/metadata/android/en-US/images/phoneScreenshots/2.png
new file mode 120000
index 0000000..8c94b0c
--- /dev/null
+++ b/fastlane/metadata/android/en-US/images/phoneScreenshots/2.png
@@ -0,0 +1 @@
+../../../../../../propaganda/Google-Play-Store/phone-01-Map.png
\ No newline at end of file
diff --git a/fastlane/metadata/android/en-US/images/phoneScreenshots/3.png b/fastlane/metadata/android/en-US/images/phoneScreenshots/3.png
new file mode 120000
index 0000000..954c0fc
--- /dev/null
+++ b/fastlane/metadata/android/en-US/images/phoneScreenshots/3.png
@@ -0,0 +1 @@
+../../../../../../propaganda/Google-Play-Store/phone-02-Map.png
\ No newline at end of file
diff --git a/fastlane/metadata/android/en-US/images/phoneScreenshots/4.png b/fastlane/metadata/android/en-US/images/phoneScreenshots/4.png
new file mode 120000
index 0000000..e497d7d
--- /dev/null
+++ b/fastlane/metadata/android/en-US/images/phoneScreenshots/4.png
@@ -0,0 +1 @@
+../../../../../../propaganda/Google-Play-Store/phone-03-Info.png
\ No newline at end of file
diff --git a/fastlane/metadata/android/en-US/images/phoneScreenshots/5.png b/fastlane/metadata/android/en-US/images/phoneScreenshots/5.png
new file mode 120000
index 0000000..9eb66f3
--- /dev/null
+++ b/fastlane/metadata/android/en-US/images/phoneScreenshots/5.png
@@ -0,0 +1 @@
+../../../../../../propaganda/Google-Play-Store/phone-04-Route.png
\ No newline at end of file
diff --git a/fastlane/metadata/android/en-US/images/phoneScreenshots/6.png b/fastlane/metadata/android/en-US/images/phoneScreenshots/6.png
new file mode 120000
index 0000000..2f6453e
--- /dev/null
+++ b/fastlane/metadata/android/en-US/images/phoneScreenshots/6.png
@@ -0,0 +1 @@
+../../../../../../propaganda/Google-Play-Store/phone-06-Nearby-framed.png
\ No newline at end of file
diff --git a/fastlane/metadata/android/en-US/images/phoneScreenshots/7.png b/fastlane/metadata/android/en-US/images/phoneScreenshots/7.png
new file mode 120000
index 0000000..233858e
--- /dev/null
+++ b/fastlane/metadata/android/en-US/images/phoneScreenshots/7.png
@@ -0,0 +1 @@
+../../../../../../propaganda/Google-Play-Store/tab-10-Map.png
\ No newline at end of file
diff --git a/fastlane/metadata/android/en-US/images/phoneScreenshots/8.png b/fastlane/metadata/android/en-US/images/phoneScreenshots/8.png
new file mode 120000
index 0000000..e378f3a
--- /dev/null
+++ b/fastlane/metadata/android/en-US/images/phoneScreenshots/8.png
@@ -0,0 +1 @@
+../../../../../../propaganda/Google-Play-Store/tab-11-Map-Flight.png
\ No newline at end of file
diff --git a/fastlane/metadata/android/en-US/images/phoneScreenshots/9.png b/fastlane/metadata/android/en-US/images/phoneScreenshots/9.png
new file mode 120000
index 0000000..76b15cf
--- /dev/null
+++ b/fastlane/metadata/android/en-US/images/phoneScreenshots/9.png
@@ -0,0 +1 @@
+../../../../../../propaganda/Google-Play-Store/tab-12-CTR.png
\ No newline at end of file
diff --git a/fastlane/metadata/android/en-US/short_description.txt b/fastlane/metadata/android/en-US/short_description.txt
new file mode 100644
index 0000000..9bad7b2
--- /dev/null
+++ b/fastlane/metadata/android/en-US/short_description.txt
@@ -0,0 +1 @@
+enroute is a free flight navigation app for VFR pilots
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index df7974a..fb018ec 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -151,6 +151,12 @@ if( ANDROID )
     # The android target require a few additional variables
     set(ANDROID_PACKAGE_SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/android" CACHE INTERNAL "")
 
+
+    message("env QTDIR               $ENV{QTDIR}")
+    message("env Qt5_DIR_ANDROID     $ENV{QTDIR}")
+    message("env QT_QMAKE_EXECUTABLE $ENV{QT_QMAKE_EXECUTABLE}")
+    message("env CMAKE_PREFIX_PATH   $ENV{CMAKE_PREFIX_PATH}")
+
     # The android target depends on our own copy of openssl
     set(ANDROID_EXTRA_LIBS
         ${openssl_SOURCE_DIR}/arm/libcrypto_1_1.so
@@ -161,6 +167,70 @@ if( ANDROID )
         ${openssl_SOURCE_DIR}/x86/libssl_1_1.so
         ${openssl_SOURCE_DIR}/x86_64/libcrypto_1_1.so
         ${openssl_SOURCE_DIR}/x86_64/libssl_1_1.so
+        $ENV{QTDIR}/qml/Qt/labs/settings/libqml_Qt_labs_settings_qmlsettingsplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/Qt/labs/settings/libqml_Qt_labs_settings_qmlsettingsplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/Qt/labs/folderlistmodel/libqml_Qt_labs_folderlistmodel_qmlfolderlistmodelplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/Qt/labs/folderlistmodel/libqml_Qt_labs_folderlistmodel_qmlfolderlistmodelplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/Qt/labs/calendar/libqml_Qt_labs_calendar_qtlabscalendarplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/Qt/labs/calendar/libqml_Qt_labs_calendar_qtlabscalendarplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtGraphicalEffects/libqml_QtGraphicalEffects_qtgraphicaleffectsplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtGraphicalEffects/libqml_QtGraphicalEffects_qtgraphicaleffectsplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtGraphicalEffects/private/libqml_QtGraphicalEffects_private_qtgraphicaleffectsprivate_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtGraphicalEffects/private/libqml_QtGraphicalEffects_private_qtgraphicaleffectsprivate_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtLocation/libqml_QtLocation_declarative_location_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtLocation/libqml_QtLocation_declarative_location_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtMultimedia/libqml_QtMultimedia_declarative_multimedia_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtMultimedia/libqml_QtMultimedia_declarative_multimedia_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtPositioning/libqml_QtPositioning_declarative_positioning_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtPositioning/libqml_QtPositioning_declarative_positioning_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQml/libqml_QtQml_qmlplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQml/libqml_QtQml_qmlplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQml/WorkerScript.2/libqml_QtQml_WorkerScript.2_workerscriptplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQml/WorkerScript.2/libqml_QtQml_WorkerScript.2_workerscriptplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQml/Models.2/libqml_QtQml_Models.2_modelsplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQml/Models.2/libqml_QtQml_Models.2_modelsplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQml/StateMachine/libqml_QtQml_StateMachine_qtqmlstatemachine_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQml/StateMachine/libqml_QtQml_StateMachine_qtqmlstatemachine_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQml/RemoteObjects/libqml_QtQml_RemoteObjects_qtqmlremoteobjects_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQml/RemoteObjects/libqml_QtQml_RemoteObjects_qtqmlremoteobjects_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQuick/Controls/libqml_QtQuick_Controls_qtquickcontrolsplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQuick/Controls/libqml_QtQuick_Controls_qtquickcontrolsplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQuick/Controls/Styles/Android/libqml_QtQuick_Controls_Styles_Android_qtquickcontrolsandroidstyleplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQuick/Controls/Styles/Android/libqml_QtQuick_Controls_Styles_Android_qtquickcontrolsandroidstyleplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQuick/Controls/Styles/Flat/libqml_QtQuick_Controls_Styles_Flat_qtquickextrasflatplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQuick/Controls/Styles/Flat/libqml_QtQuick_Controls_Styles_Flat_qtquickextrasflatplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQuick/Controls.2/Fusion/libqml_QtQuick_Controls.2_Fusion_qtquickcontrols2fusionstyleplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQuick/Controls.2/Fusion/libqml_QtQuick_Controls.2_Fusion_qtquickcontrols2fusionstyleplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQuick/Controls.2/Universal/libqml_QtQuick_Controls.2_Universal_qtquickcontrols2universalstyleplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQuick/Controls.2/Universal/libqml_QtQuick_Controls.2_Universal_qtquickcontrols2universalstyleplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQuick/Controls.2/libqml_QtQuick_Controls.2_qtquickcontrols2plugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQuick/Controls.2/libqml_QtQuick_Controls.2_qtquickcontrols2plugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQuick/Controls.2/Material/libqml_QtQuick_Controls.2_Material_qtquickcontrols2materialstyleplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQuick/Controls.2/Material/libqml_QtQuick_Controls.2_Material_qtquickcontrols2materialstyleplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQuick/Controls.2/Imagine/libqml_QtQuick_Controls.2_Imagine_qtquickcontrols2imaginestyleplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQuick/Controls.2/Imagine/libqml_QtQuick_Controls.2_Imagine_qtquickcontrols2imaginestyleplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQuick/Dialogs/libqml_QtQuick_Dialogs_dialogplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQuick/Dialogs/libqml_QtQuick_Dialogs_dialogplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQuick/Dialogs/Private/libqml_QtQuick_Dialogs_Private_dialogsprivateplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQuick/Dialogs/Private/libqml_QtQuick_Dialogs_Private_dialogsprivateplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQuick/Extras/libqml_QtQuick_Extras_qtquickextrasplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQuick/Extras/libqml_QtQuick_Extras_qtquickextrasplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQuick/Layouts/libqml_QtQuick_Layouts_qquicklayoutsplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQuick/Layouts/libqml_QtQuick_Layouts_qquicklayoutsplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQuick/PrivateWidgets/libqml_QtQuick_PrivateWidgets_widgetsplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQuick/PrivateWidgets/libqml_QtQuick_PrivateWidgets_widgetsplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQuick/Shapes/libqml_QtQuick_Shapes_qmlshapesplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQuick/Shapes/libqml_QtQuick_Shapes_qmlshapesplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQuick/Templates.2/libqml_QtQuick_Templates.2_qtquicktemplates2plugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQuick/Templates.2/libqml_QtQuick_Templates.2_qtquicktemplates2plugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQuick/Window.2/libqml_QtQuick_Window.2_windowplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQuick/Window.2/libqml_QtQuick_Window.2_windowplugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtQuick.2/libqml_QtQuick.2_qtquick2plugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtQuick.2/libqml_QtQuick.2_qtquick2plugin_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtSensors/libqml_QtSensors_declarative_sensors_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtSensors/libqml_QtSensors_declarative_sensors_arm64-v8a.so
+        $ENV{QTDIR}/qml/QtTest/libqml_QtTest_qmltestplugin_armeabi-v7a.so
+        $ENV{QTDIR}/qml/QtTest/libqml_QtTest_qmltestplugin_arm64-v8a.so
         CACHE INTERNAL ""
         )
 
@@ -168,7 +238,7 @@ if( ANDROID )
     add_library(${PROJECT_NAME} SHARED ${SOURCES} ${ANDROID_EXTRA_SOURCES})
 
     # Add libraries
-    target_link_libraries(${PROJECT_NAME} PRIVATE Qt5::AndroidExtras Qt5::Core Qt5::Positioning Qt5::Quick Qt5::Sql Qt5::Svg qhttpengine)
+    target_link_libraries(${PROJECT_NAME} PRIVATE Qt5::AndroidExtras Qt5::Core Qt5::Positioning Qt5::Quick Qt5::QuickTest Qt5::Test Qt5::Sql Qt5::Svg Qt5::Qml Qt5::QmlWorkerScript Qt5::Multimedia Qt5::MultimediaQuick Qt5::QuickShapes Qt5::RemoteObjects Qt5::Location Qt5::QuickControls2 Qt5::Sensors Qt5::SerialPort qhttpengine)
 endif()
 
 
