diff --git a/contrib/src/ffmpeg/rules.mak b/contrib/src/ffmpeg/rules.mak
index 6604a76..26ba50a 100644
--- a/contrib/src/ffmpeg/rules.mak
+++ b/contrib/src/ffmpeg/rules.mak
@@ -2,14 +2,14 @@
 
 #Uncomment the one you want
 #USE_LIBAV ?= 1
-#USE_FFMPEG ?= 1
+USE_FFMPEG ?= 1
 
 ifdef USE_FFMPEG
-FFMPEG_HASH=HEAD
+FFMPEG_HASH=n3.2.4
 FFMPEG_SNAPURL := http://git.videolan.org/?p=ffmpeg.git;a=snapshot;h=$(FFMPEG_HASH);sf=tgz
 FFMPEG_GITURL := git://git.videolan.org/ffmpeg.git
 else
-FFMPEG_HASH=HEAD
+FFMPEG_HASH=v11.9
 FFMPEG_SNAPURL := http://git.libav.org/?p=libav.git;a=snapshot;h=$(FFMPEG_HASH);sf=tgz
 FFMPEG_GITURL := git://git.libav.org/libav.git
 endif
diff --git a/contrib/src/main.mak b/contrib/src/main.mak
index 6f57cad..c6cc2d6 100644
--- a/contrib/src/main.mak
+++ b/contrib/src/main.mak
@@ -22,7 +22,7 @@ VPATH := $(TARBALLS)
 
 # Common download locations
 GNU ?= http://ftp.gnu.org/gnu
-SF := http://heanet.dl.sourceforge.net/sourceforge
+SF := http://pilotfiber.dl.sourceforge.net/sourceforge
 VIDEOLAN := http://downloads.videolan.org/pub/videolan
 CONTRIB_VIDEOLAN := http://downloads.videolan.org/pub/contrib
 
diff --git a/contrib/src/iconv/rules.mak b/contrib/src/iconv/rules.mak
index ec0f836..b45f046 100644
--- a/contrib/src/iconv/rules.mak
+++ b/contrib/src/iconv/rules.mak
@@ -6,8 +6,10 @@ PKGS += iconv
 # iconv cannot be detect with pkg-config, but it is mandated by POSIX.
 # Hard-code based on the operating system.
 ifndef HAVE_WIN32
+ifndef HAVE_ANDROID
 PKGS_FOUND += iconv
 endif
+endif
 
 $(TARBALLS)/libiconv-$(LIBICONV_VERSION).tar.gz:
 	$(call download_pkg,$(LIBICONV_URL),iconv)

