--- /home/strech/build.gradle	2020-03-30 10:40:00.000000000 -0400
+++ ./build.gradle	2020-03-30 10:45:37.772626364 -0400
@@ -8,14 +8,9 @@
         flatDir {
             dirs 'libs'
         }
-        maven {
-            url 'https://maven.fabric.io/public'
-        }
     }
     dependencies {
-        classpath 'com.google.gms:google-services:4.3.3'
         classpath 'com.android.tools.build:gradle:3.6.1'
-        classpath 'io.fabric.tools:gradle:1.31.0'
 
         // NOTE: Do not place your application dependencies here; they belong
         // in the individual module build.gradle files
@@ -26,12 +21,6 @@
     repositories {
         google()
         jcenter()
-        maven {
-            url 'https://maven.fabric.io/public'
-        }
-        maven {
-            url 'https://dl.bintray.com/rikkaw/Shizuku'
-        }
         maven { url "https://jitpack.io" }
     }
 }
