From 2e8f2ca8a86bcca829c5be4ea8ae1ea13516a549 Mon Sep 17 00:00:00 2001
From: Matthew Hague <matthewhague@zoho.com>
Date: Fri, 12 May 2017 11:32:17 +0100
Subject: [PATCH] Patch to prepare for fdroid compilation

---
 build.gradle        | 1 -
 xabber/build.gradle | 8 ++++----
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/build.gradle b/build.gradle
index 07778544..c7dfbd02 100644
--- a/build.gradle
+++ b/build.gradle
@@ -16,6 +16,5 @@ allprojects {
         jcenter()
         maven { url "https://jitpack.io" }
         maven { url 'https://oss.sonatype.org/content/repositories/snapshots' }
-        maven { url 'https://igniterealtime.org/repo' }
     }
 }
diff --git a/xabber/build.gradle b/xabber/build.gradle
index 0dddbc68..8824a6f9 100644
--- a/xabber/build.gradle
+++ b/xabber/build.gradle
@@ -92,10 +92,10 @@ dependencies {
     compile project('MemorizingTrustManager')
     compile project('emojicon')
 
-    compile "org.igniterealtime.smack:smack-android-extensions:4.2.1-20170405.080126-7"
-    compile "org.igniterealtime.smack:smack-experimental:4.2.1-20170405.080149-7"
-    compile "org.igniterealtime.smack:smack-tcp:4.2.1-20170405.080228-7"
-    compile "org.igniterealtime.smack:smack-resolver-dnsjava:4.2.1-20170405.080225-7"
+    compile "com.github.igniterealtime.smack:smack-android-extensions:$smackCommitShortHash"
+    compile "com.github.igniterealtime.smack:smack-experimental:$smackCommitShortHash"
+    compile "com.github.igniterealtime.smack:smack-tcp:$smackCommitShortHash"
+    compile "com.github.igniterealtime.smack:smack-resolver-dnsjava:$smackCommitShortHash"
 
     compile "com.android.support:design:$supportVersion"
     compile "com.android.support:support-v13:$supportVersion"
-- 
2.12.0

